CMSgov/easi-app
# Easi 2323/trb request table and gql

# EASI-2323

## Changes and Description

- Introduce TRB_Request base table, GQL, and Unit tests
     

This table will serve as the central data location for the new TRB features. There are GQL features to create, update, get an individual, and get a collection. With future TRB work, we will expand the create resolver to also create related tables. The GQL object will also expand to return logical relations 


## How to test this change
1. scripts/dev up:backend:debug && scripts/dev db:seed
2. Get Seeded Data Collection
    a. note, the below can leave out the parameter as it has a default value of false. replace with true to see archived records 
```gql
query TRBRequestGetByID{
  
  trbRequestCollection(archived: false){
    id
    name
    archived
    type
    status
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}
```
3. Update a seeded record
```gql
mutation updateTRBRequest{
  updateTRBRequest(id: "9841c768-bdcd-4856-bae2-62cfdaffacf6", changes: {
      name: "Stevens request"
      archived: true
  } ){
    id
    name
    archived
    type
    status
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}
```
4. Get a single record by ID
```GQL
query TRBRequestGetByID{
  trbRequest(id: "9841c768-bdcd-4856-bae2-62cfdaffacf6"){
    id
    name
    archived
    type
    status
    createdBy
    createdAt
    modifiedBy
    modifiedAt        
  } 
 }
```
5.  run unit tests `scripts/dev test:go`
   a. Note, resolver suite cleans the database after each test.


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
